metadata_annotator/annotate.py

```diff
@@ -311,14 +305,86 @@ def get_dimension_variables(datatree: xr.DataTree) -> set[str]:
     return dimension_variables
 
 
+def is_dimension_renaming_required(
+    var_info: VarInfoFromNetCDF4, items_to_update: set[str]
+):
+    """Determine if dimension renaming is required.
+
+    This function checks if any of the variables that have been marked for updates
+    include a `dimensions` attribute override.
+    """
+    return any(has_dimension_override(var_info, item) for item in items_to_update)
+
+
+def get_new_dimension_variables(
+    datatree: xr.DataTree, variables_to_create: set[str]
+) -> set[str]:
+    """Return the dimensions from the DataTree that are marked for creation.
+
+    For each node in the DataTree, this function checks all dataset dimensions
+    and returns those whose fully qualified paths exist in `variables_to_create`.
+    """
+    return {
+        dim_full_path
+        for node in datatree.subtree
+        for dim in node.ds.dims
+        if (dim_full_path := construct_dim_path(node.path, dim)) in variables_to_create
+    }
+
+
+def copy_shared_dimensions_to_parent(
+    datatree: xr.DataTree, variables_to_create: set[str]
+) -> None:
+    """Copy shared dimension variables to their configured parent group.
+
+    For each node in the DataTree, this function iterates over its dataset dimensions.
+    If a dimension is listed in `variables_to_create` for any parent, the dimension
+    variable is copied to the first matching parent node. This allows shared dimensions
+    to exist at higher-level groups as configured. No action is taken on dimensions that
+    are not shared between groups.
+
+    """
+    for node in datatree.subtree:
+        for dim in node.ds.dims:
+            dim_src = node.ds[dim]
+            for parent in node.parents:
+                dim_candidate_path = construct_dim_path(parent.path, dim)
+                if dim_candidate_path in variables_to_create:
+                    parent.ds = parent.ds.assign(**{dim: dim_src})
```

Comment:
do you need to unpack that dict? couldn't you just assign it as a dict? 
```suggestion
                    parent.ds = parent.ds.assign({dim: dim_src})
```

Reply:
Nice catch. Updated in [ae7d2bf](https://github.com/nasa/harmony-metadata-annotator/pull/36/commits/ae7d2bf9488c0f86f3491d9b20b845ad7ab5071b).